.github/workflows/publish-docker-main.yml

```
@@ -52,17 +51,13 @@ jobs:
         run: make generate
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
-      - name: Build docker image
-        run: |
-          make docker || make docker
       - name: Log in to the Container registry
         uses: docker/login-action@v1.10.0
-        if: github.ref == 'refs/heads/main'
         with:
           registry: ${{ env.HUB }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Push docker image
-        if: github.ref == 'refs/heads/main'
+      - name: Build and Push docker image
         run: |
-          make docker.push || make docker.push
\ No newline at end of file
+          make docker || make docker
+          make docker.push || make docker.push
```

> Also, separating the `publish-docker` workflow into 2 files makes no sense IMO, the reason why we want the workflow to run (partially though) in PRs is that we want to check the commands in that workflow is runnable and don't want it to break after merging. Separating it into 2 files may lead to a situation where developers adjust the file `.github/workflows/publish-docker-pr.yml` to make the CI pass and merge the PRs, but forget to adjust `.github/workflows/publish-docker-main.yml`.

What I suggested is something like this:

```suggestion
          make docker || make docker
          if [[ ${{ github.ref == 'refs/heads/main' ]]; then
            make docker.push || make docker.push
          fi
```